src/interface/SpellIcon.tsx

```diff
@@ -29,7 +29,7 @@ const SpellIcon = ({ spell, noLink, alt, ilvl, ...others }: Props) => {
     />
   );
 
-  const spellId = typeof spell === 'number' ? spell : spell.id;
+  const spellId = typeof spell === 'number' ? spell : spell?.id;
```

Comment:
not sure what was being worked around by this, but this is not the correct fix. as per the type: `spell` is a `number` or `Spell`. it should never be null/undefined---although that can happen due to some JS code. the correct fix is to fix the component that is passing `null`/`undefined` here.

im assuming this was just some old `Analyzer` passing in bad data from a spell that no longer exists and probably won't trigger again if it this change is reverted, but ill double check.

EDIT: don't see any broken places in MM now with this removed. if there are specific ones, I can take a look at fixing the callsites.

Reply:
I'm not encountering the issue either now. Maybe it was a weird compilation problem. All good with this revert.